### ISSUE TITLE
FormGroup inside Modal got rendered before Modal is open

### ISSUE THREAD

**Issue by yaooluu** [vmware/clarity#1557]
**Select one ...**  (check one with "x")
```
[x] bug
[ ] feature request
[ ] enhancement
```

### Expected behavior
We have a very common scenario of using FormGroup inside Modal. We use "[(clrModalOpen)]" two way binding to control the visibility of modal. We initialize the FormGroup only before we open the modal, not during ngOnInit().

According from the code below, I'm expecting "clrModalOpen" to be the "Clarity" way of using *ngIf on a modal.
https://github.com/vmware/clarity/blob/16658ecf565f9c86a368b1574dde3d06c0219b1d/src/clarity-angular/modal/modal.html#L8

### Actual behavior
The modal content is being rendered even before modal is opened, causing Angular giving the following error:

> formGroup expects a FormGroup instance. Please pass one in.

The root cause might be related to  https://github.com/vmware/clarity/issues/1537

### Reproduction of behavior
https://plnkr.co/edit/VbDSzz?p=preview

One workaround is initialize FormGroup with empty object during declarition. But this approach is against the whole purpose of delaying the creation of FormGroup.

> myFormGroup = new FormGroup({});

Another workaround is adding *ngIf. **But I think this is highly discouraged by Clarity.**
> <clr-modal [(clrModalOpen)]="open" *ngIf="open">

### Environment details

* **Angular version:** 4.3.0

* **Clarity version:** 0.10.2

* **OS and version:** Mac OS El Captain

* **Browser:** Chrome 61

**Comment by mathisscott** [comment]
@youdz : Could you advise on whether or not `*ngIf` is the right approach?

**Comment by youdz** [comment]
Actually, that `*ngIf` solution is currently the recommended workaround, so feel free to go ahead and use it.

This is a known issue, it was first described in #215, and we planned to work on it as part of #903 (the actual issue is #895 but the description is in the epic). To understand exactly what is happening, I recommend reading https://medium.com/claritydesignsystem/ng-content-the-hidden-docs-96a29d70d11b (shameless plug).

The problem we have at the moment is that the API and implementation we plan on doing is currently blocked by https://github.com/angular/angular/issues/15998. That issue was just assigned a priority 5 last week, so hopefully it'll be fixed anytime now.

I'm closing this as a duplicate of the various issues linked here, but thanks for reporting this.
